web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -74,58 +75,63 @@ public JsPartitionedTable(WorkerConnection connection, JsWidget widget) {
 
     @JsIgnore
     public Promise<JsPartitionedTable> refetch() {
-        if (keys != null) {
-            keys.close();
-        }
-        if (subscription != null) {
-            subscription.close();
-        }
+        closeSubscriptions();
         return widget.refetch().then(w -> {
             descriptor = PartitionedTableDescriptor.deserializeBinary(w.getDataAsU8());
 
             return w.getExportedObjects()[0].fetch();
-        }).then(result -> {
-            keys = (JsTable) result;
+        }).then(result -> connection.newState((c, state, metadata) -> {
+            rawKeyTable = (JsTable) result;
```

Comment:
Why do we need to retain this as a field at all? Can't we just close it in the next .then() down, after this operation was successful and we have the keyTable?

Reply:
It was annoying with closures, the `.then` doesn't have a ref to the table. I changed to `.close` after calling the `.dropColumns`, is that safe? Otherwise I'll refactor to store it in a 1-element array at the top of the function.

Reply:
One element array is a classic/terrible hack, so is using `AtomicReference` to point to the `final` instance.